File: ItemCount/Module/VB/ItemCount.Module/Controllers/RefreshNavigationController.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.SystemModule

Public Class RefreshNavigationController
    Inherits ViewController
    Public Sub New()
        TargetViewNesting = Nesting.Root
        TargetObjectType = GetType(Email)
    End Sub
    Protected Overrides Sub OnActivated()
        MyBase.OnActivated()
        AddHandler View.ObjectSpace.Committed, AddressOf ObjectSpace_Committed
    End Sub

    Private Sub ObjectSpace_Committed(ByVal sender As Object, ByVal e As System.EventArgs)
        Dim controller As ShowNavigationItemController = Application.MainWindow.GetController(Of ShowNavigationItemController)()

        If controller IsNot Nothing Then
            controller.ShowNavigationItemAction.BeginUpdate()
            controller.RecreateNavigationItems()
            controller.ShowNavigationItemAction.EndUpdate()
        End If
    End Sub

    Protected Overrides Sub OnDeactivated()
        RemoveHandler View.ObjectSpace.Committed, AddressOf ObjectSpace_Committed
        MyBase.OnDeactivated()
    End Sub
End Class

File: CS/EFCore/ItemCountEF/ItemCountEF.Module/Controllers/RefreshNavigationController.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.SystemModule;
using dxTestSolution.Module.BusinessObjects;

namespace ItemCount.Module.Controllers {
    public class RefreshNavigationController : ViewController {
        public RefreshNavigationController() {
            TargetViewNesting = Nesting.Root;
            TargetObjectType = typeof(Email);
        }
        protected override void OnActivated() {
            base.OnActivated();
            View.ObjectSpace.Committed += ObjectSpace_Committed;
        }
        private void ObjectSpace_Committed(object sender, System.EventArgs e) {
            ShowNavigationItemController controller = Application.MainWindow.GetController<ShowNavigationItemController>();
            if(controller != null) {
                controller.ShowNavigationItemAction.BeginUpdate();
                controller.RecreateNavigationItems();
                controller.ShowNavigationItemAction.EndUpdate();
            }
        }
        protected override void OnDeactivated() {
            View.ObjectSpace.Committed -= ObjectSpace_Committed;
            base.OnDeactivated();
        }
    }
}
